scripts/prepare_sector_network.py

```diff
@@ -157,6 +157,113 @@ def add_generation(
         # set the "co2_emissions" of the carrier to 0, as emissions are accounted by link efficiency separately (efficiency to 'co2 atmosphere' bus)
         n.carriers.loc[carrier, "co2_emissions"] = 0
 
+    # add coal CC technologies
+    if options["coal"].get("CC", False):
+        coal_CC_techs = options["coal"].get("coal_CC_techs", list())
+
+        coal_CC_names = {
+            "coal CC-95": "Coal-95%-CCS",
+            "coal CC-99": "Coal-99%-CCS",
+            "coal IGCC CC-90": "Coal-IGCC-90%-CCS",
+        }
+
+        for cc_tech in coal_CC_techs:
+            n.madd(
+                "Link",
+                spatial.nodes + " " + cc_tech,
+                bus0=spatial.coal.nodes,
+                bus1=spatial.nodes,
+                bus2="co2 atmosphere",
+                bus3=spatial.co2.nodes,
+                marginal_cost=costs.at[coal_CC_names[cc_tech], "efficiency"]
+                * costs.at[coal_CC_names[cc_tech], "VOM"],  # NB: VOM is per MWel
+                # NB: fixed cost is per MWel
+                capital_cost=costs.at[coal_CC_names[cc_tech], "efficiency"]
+                * costs.at[coal_CC_names[cc_tech], "fixed"],

```

Comment:
Yes, this is what we have discussed with @danielelerede-oet. In PyPSA-Eur, they consider capital cost of coal CC constitute from investment costs for coal part and capture part. For capture, the only available capture technology is costs is `biomass CHP capture`. However, here we directly used cost parameters of coal CC technologies, because NREL ATB already contains investment costs for technologies with capture (no need to separately account for coal and carbon capture, it is combined).

Reply:
Thanks a lot for looking into these details. As far as I'm aware, biomass co-firing in CHPs is a "soft" way to decrease CO2 emissions for coal-fired CHPs while keeping the costs low. However, there are some controversies around that (in particular to what relates for food-energy nexus and CO2 capture rates), so PyPSA-Eur implementation seems to be not a very general one, and most likely linked with some particular project

So, completely support the design you have selected